src/libraries/Common/src/System/HexConverter.cs

```diff
@@ -84,15 +92,84 @@ public static void ToCharsBuffer(byte value, Span<char> buffer, int startingInde
             buffer[startingIndex] = (char)(packedResult >> 8);
         }
 
+#if !NETFRAMEWORK && !NETSTANDARD1_0 && !NETSTANDARD1_3 && !NETSTANDARD2_0 && NETCOREAPP
+    public static void EncodeToUtf16_Ssse3(ReadOnlySpan<byte> bytes, Span<char> chars, Casing casing = Casing.Upper)
+    {
+        int pos = 0;
+
+        Vector128<byte> shuffleMask = Vector128.Create(
+            0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
+            0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF);
+
+        Vector128<byte> asciiTable = (casing == Casing.Upper) ?
+            Vector128.Create((byte)'0', (byte)'1', (byte)'2', (byte)'3',
```

Comment:
Out of curiosity - is there a perf difference between using `Vector128.Create` inline here vs. defining it as a static readonly field and doing a glorified `ldsfld` on that field?

Reply:
@GrabYourPitchforks 
So there are 3 patterns we currently use in BCL for const vectors:

```csharp
    // Case 1: Plain Vector.Create for mask

    static Vector128<byte> Case1(Vector128<byte> vec)
    {
        return Ssse3.Shuffle(vec, Vector128.Create(
            0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
            0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF));
    }

    // Case 2: static readonly Vector field for mask

    private static readonly Vector128<byte> s_vec = Vector128.Create(
        0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
        0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF);
    static Vector128<byte> Case2(Vector128<byte> vec)
    {
        return Ssse3.Shuffle(vec, s_vec);
    }

    // Case 3: Roslyn's hack

    private static ReadOnlySpan<byte> ShuffleMaskSpan => new byte[]{
        0xFF, 0xFF, 0, 0xFF, 0xFF, 0xFF, 1, 0xFF,
        0xFF, 0xFF, 2, 0xFF, 0xFF, 0xFF, 3, 0xFF};
    static Vector128<byte> Case3(Vector128<byte> vec)
    {
        return Ssse3.Shuffle(vec, Unsafe.ReadUnaligned<Vector128<byte>>(
            ref MemoryMarshal.GetReference(ShuffleMaskSpan)));
    }
```

Here is ASM for them:
```ASM
; Method Case1
G_M42977_IG01:
       vzeroupper 
G_M42977_IG02:
       vmovupd  xmm0, xmmword ptr [rdx]
       vpshufb  xmm0, xmm0, xmmword ptr [reloc @RWD00]
       vmovupd  xmmword ptr [rcx], xmm0
       mov      rax, rcx
G_M42977_IG03:
       ret      
RWD00  	dq	FF01FFFFFF00FFFFh, FF03FFFFFF02FFFFh
; Total bytes of code: 24


; Method Case2
G_M48322_IG01:
       push     rsi
       sub      rsp, 48
       vzeroupper 
       mov      rsi, rcx
G_M48322_IG02:
       vmovupd  xmm0, xmmword ptr [rdx]
       vmovupd  xmmword ptr [rsp+20H], xmm0
       mov      rcx, 0xD1FFAB1E
       mov      edx, 2
       call     CORINFO_HELP_GETSHARED_NONGCSTATIC_BASE
       mov      rax, 0xD1FFAB1E
       mov      rax, gword ptr [rax]
       vmovupd  xmm0, xmmword ptr [rsp+20H]
       vpshufb  xmm0, xmm0, xmmword ptr [rax+8]
       vmovupd  xmmword ptr [rsi], xmm0
       mov      rax, rsi
G_M48322_IG03:
       add      rsp, 48
       pop      rsi
       ret      
; Total bytes of code: 79


; Method Case3
G_M55587_IG01:
       vzeroupper 
G_M55587_IG02:
       mov      rax, 0xD1FFAB1E
       vmovupd  xmm0, xmmword ptr [rax]
       vmovupd  xmm1, xmmword ptr [rdx]
       vpshufb  xmm0, xmm1, xmm0
       vmovupd  xmmword ptr [rcx], xmm0
       mov      rax, rcx
G_M55587_IG03:
       ret      
; Total bytes of code: 34
```
As you can see, the 1st case is the best one - JIT saved vector into the data section and furthermore - sometimes it's able to load it as part of the instruction needed as mem op without additional registers. Also, JIT now can do CSE for them. Thus we can probably do some clean up across the BCL. The only issue we still have - such constants aren't hoisted from loops by JIT yet.

Reply:
That's nifty! Thanks for the detailed explanation. :)

Reply:
>> So there are 3 patterns we currently use in BCL for const vectors

> As you can see, the 1st case is the best one - JIT saved vector into the data section and furthermore - sometimes it's able to load it as part of the instruction we use that mask for (as a mem op without additional registers). Also, JIT now can do CSE for them. Thus we can probably do some clean up across the BCL.

Sounds like a good thing to do. Can you open an issue? Thanks

Reply:
Filed https://github.com/dotnet/runtime/issues/44115